parser/mpFuncCommon.cpp

```diff
@@ -421,12 +421,16 @@ MUP_NAMESPACE_START
 
     // Matches exactly: "yyyy-mm-dd"
     std::regex basic_date ("^\\d{4}-\\d{2}-\\d{2}$");
-
-    if (std::regex_match (date_a, basic_date)) {
+    if (std::regex_match (date_a, basic_date) && std::regex_match (date_b, basic_date)) {
       date_a = date_a + "T00:00";
-    }
-    if (std::regex_match (date_b, basic_date)) {
       date_b = date_b + "T00:00";
+    } else if (std::regex_match (date_a, basic_date) || std::regex_match (date_b, basic_date)) {

```

Comment:
hmm! Would not be better to verify if the format is including date and time instead accept that?